### PR TITLE
Add Cloud Run Operator

### PR DESCRIPTION
**Pull request by mattsax**
This PR add a `CloudRunOperator` along with `CloudRunHook`.
This new Operator allows to make authenticated call against Cloud Run (HTTP request with proper authentication header)

`CloudRunOperator` extends the `SimpleHttpOperator`: this add a dependency to HTTP provider.
The authentication token needed in HTTP header (for proper Cloud Run authentication) is an [ID Token](https://cloud.google.com/docs/authentication/get-id-token).
The `credential_provider.py` had to be modified with a new function `get_id_token_credentials` to provide "ID Token like" credentials.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/utils/credentials_provider.py [hunk 1]
```diff
@@ -332,6 +356,16 @@ def get_credentials_and_project_id(*args, **kwargs) -> tuple[google.auth.credent
     return _CredentialProvider(*args, **kwargs).get_credentials_and_project()
 
 
+def get_id_token_credentials(*args, **kwargs) -> tuple[google.auth.credentials.Credentials]:
```
Comment (mik-laj) [inline airflow/providers/google/cloud/utils/credentials_provider.py:359]: I wonder how much this code has to do with the this code:
https://github.com/apache/airflow/blob/0fd129f33792a95b06c4f4ad1f7ee83dc2f5cef5/airflow/providers/google/common/utils/id_token_credentials.py

Comment (mattsax) [reply airflow/providers/google/cloud/utils/credentials_provider.py:359]: The two seems indeed related. But `id_token_credentials.py` provide the credentials from [Google application default credentials](https://cloud.google.com/docs/authentication/application-default-credentials).
Whereas the `credential_provider.py` already contains all the code to infer credentials from an Airflow Connection

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Static checks failing.

**Comment by mattsax** [comment]
I fixed the doc Build.
But for the static checks, the failure comes from `airflow/providers/neo4j/hooks/neo4j.py`. It doesn't seem related to this PR.

**Comment by potiuk** [comment]
Conflicts need to be solved.

**Comment by potiuk** [comment]
And errors to fix.

**Comment by eladkal** [comment]
I rebased the PR. lets see if test passes

**Comment by steren** [comment]
Hi, I am the Product Manager for Cloud Run. Thanks for working on a Cloud Run Operator for Airflow!

Please feel free to ask any Cloud Run related question here.

I looked at the PR, it seems that it is using an HTTP call to a Cloud Run service. I would have believed that a [Cloud Run job](https://cloud.google.com/run/docs/overview/what-is-cloud-run#jobs) would be more suited for Airflow, as it doesn't require the container to expose any HTTP server, it can be executed via a simple API call. Have you evaluated a job?

**Comment by matthieuvilatte** [comment]
Hi @steren 
Depending on the use case we use both Cloud Run Service and Cloud Run Jobs.
(An other PR has been submited to add a [Cloud Run Job operator](https://github.com/apache/airflow/pull/28525) )
I deployed [dbt](https://www.getdbt.com/) using Cloud Run. It is pretty convient to be abble to send parameters though a POST HTTP request: `start_date` and `end_date` or even `--full-refresh` parameter when building an incremental model.

**Comment by steren** [comment]
@matthieuvilatte Ah I might have misunderstood this Operator. It doesn't deploy a Cloud Run service but calls an existing Cloud Run service, is that right?

Maybe the Operator name should make this clearer (i.e. Cloud Run hook)? To differentiate it with the other "Cloud Run Job" Operator

**Comment by matthieuvilatte** [comment]
@steren 
This Operator doesn't deploy a new Cloud Run Service, but can be used to make an authenticated call to an exising Cloud Run Service using a Service Account stored in Airflow Connections.

This Operator name is indeed confusing, but I don't think we should rename it `Cloud Run hook`. I might be wrong but when describing Airflow DAGs, we always use Operators. An Operator holds the `execute` method which is called when executing the task and uses a hook to deal with the connection.

May be `Cloud Run Service Call Operator`?

**Comment by eladkal** [comment]
@matthieuvilatte operators usually design to do a specific action thus we like/prefer when operator has a verb in it's name indicate the action for example: `SqlExecuteOperator`, `EC2StartInstanceOperator`, `WasbDeleteBlobOperator`

**Comment by eladkal** [comment]
Tests are failing.
Could you please fix it?

**Pull request closed without merging**